src/meta/src/barrier/mod.rs

```diff
@@ -489,7 +496,7 @@ impl GlobalBarrierManager {
     }
 
     /// Check whether we should pause on bootstrap from the system parameter and reset it.
-    async fn take_pause_on_bootstrap(&self) -> MetaResult<bool> {
+    async fn take_pause_on_bootstrap(&mut self) -> MetaResult<bool> {
```

Comment:
What's this change for?

Reply:
Interesting. This sounds like the problem that [`Exclusive`](https://doc.rust-lang.org/stable/std/sync/struct.Exclusive.html) is designed for.